src/main/java/io/github/railroad/project/ui/welcome/WelcomeProjectsPane.java

```diff
@@ -46,27 +58,39 @@ public WelcomeProjectsPane() {
             } else if (event.getCode() == KeyCode.DELETE) {
                 Project project = projectsList.getSelectionModel().getSelectedItem();
                 if (project != null) {
-                    project.delete(false); // TODO: Ask for confirmation
-                    removeProject(project);
+                    manager.RemoveProject(project);
+                    filterProjects("");
                 }
 
                 event.consume();
             }
         });
 
-        projectsList.getItems().addAll(Project.loadProjects());
-
+        filterProjects("");
         setContent(projectsList);
     }
 
-    public void addProject(Project project) {
-        projectsList.getItems().add(project);
-    }
-
     public void removeProject(Project project) {
         projectsList.getItems().remove(project);
     }
 
+    public void filterProjects(String value) {
+        projectsList.getItems().clear();
+        if (value == null || value.isEmpty()) {
+            projectsList.getItems().addAll(manager.getProjectCollection());
+        } else {
+            List<Project> filteredProjects = new ArrayList<>();
+
+            for (Project project : manager.getProjectCollection()) {
+                if (project.getAlias().toLowerCase().contains(value.toLowerCase())) {
+                    filteredProjects.add(project);
+                }
+            }
+            projectsList.getItems().addAll(filteredProjects);
+        }
+    }
+
+
```

Comment:
```suggestion
    public void filterProjects(String value) {
        projectsList.getItems().clear();
        
        if (value == null || value.isEmpty()) {
            projectsList.getItems().addAll(manager.getProjectCollection());
            return;
        }
        
        List<Project> filteredProjects = new ArrayList<>();
        for (Project project : manager.getProjectCollection()) {
            if (project.getAlias().toLowerCase(Locale.ROOT).contains(value.toLowerCase(Locale.ROOT))) {
                filteredProjects.add(project);
            }
        }
        
        projectsList.getItems().addAll(filteredProjects);
    }
```
Just some small cleanups:
- Whitespace
- Using `Locale.ROOT` when doing `toLowerCase` and `toUpperCase`
- Terminating instead of having an else statement is a good way to avoid nesting and should be done wherever it makes sense to do so (here)